sw/host/ot_certs/src/asn1/der.rs

```diff
@@ -158,6 +158,31 @@ impl Builder for Der {
         })
     }
 
+    fn push_bitstring(
+        &mut self,
+        name_hint: Option<String>,
+        tag: &Tag,
+        bits: &[Value<bool>],
+    ) -> Result<()> {
+        let bits = bits
+            .iter()
+            .map(Self::get_value_or_error)
+            .collect::<Result<Vec<_>>>()?;
+        // See X.690 spec section 8.6 for encoding details.
+        // Note: the encoding of an empty bitstring must be the number of unused bits to 0 and have no content.
+        let nr_bytes = (bits.len() + 7) / 8;
+        let mut bytes = vec![0u8; nr_bytes];
+        for (i, bit) in bits.iter().enumerate() {
+            if **bit {
+                bytes[i / 8] |= 1u8 << (7 - (i % 8));
+            }
+        }
```

Comment:
```suggestion
        for (i, bit) in bits.into_iter().enumerate() {
            bytes[i / 8] |= (*bit as u8) << (7 - (i % 8));
        }
```
`into_iter` will consume bits so you'll need to create a variable to store bits.len() for later use.